pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,573 @@
+    )
+
+    voc_setpoint = Instrument.control(
+        ":BATT:SIM:VOC?",
+        ":BATT:SIM:VOC %g",
+        """Control the Voc of the simulated battery.""",
+        validator=truncated_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    voc_full = Instrument.control(
+        ":BATT:SIM:VOC:FULL?",
+        ":BATT:SIM:VOC:FULL %g",
+        """Control the Voc for the full simulated battery.""",
+        validator=truncated_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    voc_empty = Instrument.control(
+        ":BATT:SIM:VOC:EMPT?",
+        ":BATT:SIM:VOC:EMPT %g",
+        """Control the Voc for the empty simulated battery.""",
+        validator=truncated_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    _query_buffer_data = Instrument.measurement(
+        ':BATT:DATA:DATA? "VOLT, CURR, SOC, RES,REL"',
+        """Get the buffer in battery simulator mode""",
+        separator=",",
+    )
+
+    @property
+    def buffer_data(self) -> pd.DataFrame:
+        """Get the buffer in battery simulator mode and return its content as a pandas dataframe"""
+        if not self.parent.reading_available:
+            return pd.DataFrame(
+                {"current": [], "voltage": [], "soc": [], "resistance": [], "time": []}
+            )
+        data = np.array(self._query_buffer_data, dtype=np.float32)
+        return pd.DataFrame(
+            {
+                "voltage": data[0::5],
+                "current": data[1::5],
+                "soc": data[2::5],
+                "resistance": data[3::5],
+                "time": data[4::5],
+            }
+        )
+
+
+class Keithley2281S(SCPIMixin, Instrument, KeithleyBuffer):
+    """
+    Represents the Keithley 2281S-20-6 power supply and battery simulator / characterizer.
+    Common commands beside `function_mode` and power supply commands should also work for
+    Keithley 2280S power supplies, although this is untested.
+    """
+
+    ps = Instrument.ChannelCreator(PowerSupplyChannel, "PS")
+    bt = Instrument.ChannelCreator(BatteryTestChannel, "BT")
+    bs = Instrument.ChannelCreator(BatterySimulationChannel, "BS")
+
+    def __init__(self, adapter, name="Keithley2281S", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+        if self.line_frequency == 60:
+            _PLC_RANGE[1] = 15
+
+    # Overwritten parent property
+
+    @property
+    def buffer_data(self):
+        """Get a pandas dataframe of values from the buffer."""
+        # Replace with match case, once Python>=3.10 is required
+        function_mode = self.function_mode
+        if function_mode == "POWER":
+            return self.ps.buffer_data
+        if function_mode == "TEST":
+            return self.bt.buffer_data
+        if function_mode == "SIMULATOR":
+            return self.bs.buffer_data
+
+    display_text_data = Instrument.setting(
+        ":DISP:USER:TEXT '%s'", """Set control text to be displayed(24 characters)."""
+    )
+
+    function_mode = Instrument.control(
```

Comment:
What about a method for each channel, which enables that channel. E.g. `enable_channel`, such that you start using a channel with the channel itself?

Reply:
In this case i think it is clearer to have the `function_mode` in the parent device, since the channels are not actual channels, but groups of functionalities that are mutually exclusive. I added an explanation to the docstring of each channel.  
For the other comments i implemented the review suggestions.

Reply:
Ok.